nixos/modules/config/update-users-groups.pl

```diff
@@ -3,22 +3,25 @@
 use File::Slurp;
 use JSON;
 
-make_path("/var/lib/nixos", { mode => 0755 });
-
-
 # Keep track of deleted uids and gids.
 my $uidMapFile = "/var/lib/nixos/uid-map";
 my $uidMap = -e $uidMapFile ? decode_json(read_file($uidMapFile)) : {};
 
 my $gidMapFile = "/var/lib/nixos/gid-map";
 my $gidMap = -e $gidMapFile ? decode_json(read_file($gidMapFile)) : {};
 
+my $is_dry = $ENV{'NIXOS_ACTION'} // "" eq "dry-activate";
```

Comment:
A suggestion to support `perl update-users-groups.pl foo.json --dry`

```suggestion
use Getopt::Long;

my $is_dry = $ENV{'NIXOS_ACTION'} // "" eq "dry-activate";
GetOptions("dry-activate" => \$is_dry);

warn "warning: dry activation\n" if $is_dry;

my $spec = decode_json(read_file($ARGV[0]));

```